ctapipe/image/toymodel.py

```diff
@@ -40,35 +41,33 @@
 TOYMODEL_RNG = default_rng(0)
 
 
-@u.quantity_input(
-    x=u.m,
-    y=u.m,
-    centroid_x=u.m,
-    centroid_y=u.m,
-    psi=u.deg,
-    time_gradient=u.ns / u.m,
-    time_intercept=u.ns,
-)
+def _obtain_image_no_units(
+    x, y, centroid_x, centroid_y, psi, time_gradient, time_intercept
+):
+    longitudinal, _ = camera_to_shower_coordinates(x, y, centroid_x, centroid_y, psi)
+    return longitudinal * time_gradient + time_intercept
+
+
 def obtain_time_image(x, y, centroid_x, centroid_y, psi, time_gradient, time_intercept):
     """Create a pulse time image for a toymodel shower. Assumes the time development
     occurs only along the longitudinal (major) axis of the shower, and scales
     linearly with distance along the axis.
 
     Parameters
     ----------
-    x : u.Quantity[length]
+    x : u.Quantity[angle|length]
         X camera coordinate to evaluate the time at.
         Usually the array of pixel X positions
-    y : u.Quantity[length]
+    y : u.Quantity[angle|length]
         Y camera coordinate to evaluate the time at.
         Usually the array of pixel Y positions
-    centroid_x : u.Quantity[length]
+    centroid_x : u.Quantity[angle|length]
         X camera coordinate for the centroid of the shower
-    centroid_y : u.Quantity[length]
+    centroid_y : u.Quantity[angle|length]
         Y camera coordinate for the centroid of the shower
     psi : convertible to `astropy.coordinates.Angle`
         rotation angle about the centroid (0=x-axis)
-    time_gradient : u.Quantity[time/length]
+    time_gradient : u.Quantity[time/angle]
```

Comment:
What exactly does this mean? Time divided by length|angle? That would be smth like `time/[angle|length]` maybe? 
Or can we get away with just writing `length` everywhere and in the docstring note that the unit of a length can be that of a distance or an angle?

Reply:
that might be the better way, yes.